src/main.rs

```diff
@@ -0,0 +1,57 @@
+use rand::Rng;
+use std::fmt;
+
+fn main() {
+    let mut bag = TileBag::new_default();
+    for i in 0..1000 {
+        let tile = bag.draw_tile();
+        println!("{}", tile)
+    }
+}
+
+struct TileBag {
+    bag: Vec<char>,
+    rng: rand::rngs::ThreadRng,
+}
+
+impl TileBag {
+    fn new(letter_distribution: [u8; 26]) -> Self {
+        let mut bag: Vec<char> = vec![];
+        let range: std::ops::Range<u8> = std::ops::Range { start: 0, end: 26 };
+        for letter in range {
+            for i in 0..letter_distribution[letter as usize] {
+                bag.push((letter + 65) as char);
+            }
+        }
+
+        TileBag {
+            bag: bag,
+            rng: rand::thread_rng(),
+        }
+    }
+
+    fn new_default() -> Self {
+        Self::new([
+            // banagrams letter distribution
+            13, 3, 3, 6, 18, 3, 4, 3, 12, 2, 2, 5, 3, 8, 11, 3, 2, 9, 6, 9, 6, 3, 3, 2, 3, 2,
+        ])
+    }
+
+    // TODO: get O(1) performance on draw - maybe a hashmap?
```

Comment:
https://doc.rust-lang.org/std/collections/struct.HashSet.html

HashSet could be one to look into as it could also provide protection for returning tiles that weren't originally in the bag depending on its structure

Reply:
It looks like it's tough to randomly sample from a HashSet, and it doesn't allow duplicate keys, so you could store multiple `A` characters etc. I think Vec may be right, just need something better than `remove`, which says it can be `O(n)`.

Reply:
In fact, it looks like `swap_remove` is exactly the method I was looking for.